src/client/Microsoft.Identity.Client/ManagedIdentity/V2/CsrMetadata.cs

```diff
@@ -7,18 +7,24 @@
     using Microsoft.Identity.Json;
 #endif
 
-namespace Microsoft.Identity.Client.ManagedIdentity
+namespace Microsoft.Identity.Client.ManagedIdentity.V2
 {
     /// <summary>
     /// Represents VM unique Ids for CSR metadata.
     /// </summary>
     internal class CuidInfo
     {
-        [JsonProperty("vmid")]
-        public string Vmid { get; set; }
+        [JsonProperty("vmId")]
+        public string VmId { get; set; }
 
-        [JsonProperty("vmssid")]
-        public string Vmssid { get; set; }
+        [JsonProperty("vmssId")]
+        public string VmssId { get; set; }
+
+        public static bool IsNullOrEmpty(CuidInfo cuidInfo)
+        {
+            return cuidInfo == null ||
+                   (string.IsNullOrEmpty(cuidInfo.VmId) && string.IsNullOrEmpty(cuidInfo.VmssId));
```

Comment:
suggest testing this against the newly created VM. It doesn't return a VMSSID when I checked

Reply:
This correctly ensures Cuid is not empty